### ISSUE TITLE
login

### ISSUE THREAD

**Issue by nishitjpatel** [statusengine/interface#25]
I need help getting basic auth to work with statusengineUI.   My next goal would be setup ldap auth if i can get basic auth going.    When i goto statusengine page there is nothing.   i enabled allow_anonymous and it had zero effect. 

Notice below error message in /var/log/httpd/
AH00130: File does not exist: /usr/share/statusengine-ui/public/api/index.php/login

statusengine worker is good.

# /opt/statusengine/worker/bin/Console.php statistics
Statusengine statistics @16:23:47
Cluster node: Crowbar

Live data hold in Redis:
===================================
Total processed host status records: 2
Total processed host status records last minute: 2
Total processed service status records: 10
Total processed service status records last minute: 0

Processed Historical data:
===================================
Total processed host check records: 1
Total processed host check records last minute: 1
Total processed service check records: 5
Total processed service check records last minute: 0
Total processed log entry records: 0
Total processed log entry records last minute: 0
Total processed state change records: 0
Total processed state change records last minute: 0
Total processed performance data records: 0
Total processed performance data records last minute: 0

Processed Misc data:
===================================
Total processed Notifications, Downtimes, Acknowledgements, etc: 0
Total processed Notifications, Downtimes, Acknowledgements, etc last minute: 0

Overall:
===================================
Total processed records: 18
Total processed records last minute: 3

Statusengine process intormation:
===================================
Last update of this statistics: 16:23:47 - 25.07.2019
Number of workers: 7
Total number of running Statusengine processes: 8
Statusengine is running since: : 2 minutes and 18 seconds

I read another post about basic auth and saw the other user didnt have php install, not the case here

# rpmquery -a |grep php
php-PsrLog-1.1.0-1.el7.noarch
php-symfony-yaml-2.8.12-2.el7.noarch
php-symfony-browser-kit-2.8.12-2.el7.noarch
php-composer-semver-1.5.0-1.el7.noarch
php-symfony-event-dispatcher-2.8.12-2.el7.noarch
php-intl-7.3.7-3.el7.remi.x86_64
php-pecl-gearman-2.0.5-3.el7.remi.7.3.x86_64
php-symfony-process-2.8.12-2.el7.noarch
php-jsonlint-1.6.1-1.el7.noarch
php-composer-xdebug-handler-1.3.3-1.el7.noarch
php-pear-1.10.9-3.el7.remi.noarch
php-json-7.3.7-3.el7.remi.x86_64
php-mysqlnd-7.3.7-3.el7.remi.x86_64
php-symfony-class-loader-2.8.12-2.el7.noarch
php-symfony-dependency-injection-2.8.12-2.el7.noarch
php-paragonie-random-compat-2.0.18-1.el7.noarch
php-gd-7.3.7-3.el7.remi.x86_64
php-ldap-7.3.7-3.el7.remi.x86_64
php-symfony-filesystem-2.8.12-2.el7.noarch
php-symfony-dom-crawler-2.8.12-2.el7.noarch
php-composer-spdx-licenses-1.5.1-1.el7.noarch
php-symfony-console-2.8.12-2.el7.noarch
php-pecl-igbinary-3.0.1-2.el7.remi.7.3.x86_64
php-cli-7.3.7-3.el7.remi.x86_64
php-bcmath-7.3.7-3.el7.remi.x86_64
php-justinrainbow-json-schema5-5.2.8-1.el7.noarch
php-seld-phar-utils-1.0.1-1.el7.noarch
php-symfony-http-foundation-2.8.12-2.el7.noarch
php-common-7.3.7-3.el7.remi.x86_64
php-pecl-redis4-4.3.0-2.el7.remi.7.3.x86_64
php-7.3.7-3.el7.remi.x86_64
php-symfony-config-2.8.12-2.el7.noarch
php-symfony-expression-language-2.8.12-2.el7.noarch
php-password-compat-1.0.4-1.el7.noarch
php-symfony-http-kernel-2.8.12-2.el7.noarch
php-pecl-zip-1.15.4-1.el7.remi.7.3.x86_64
php-process-7.3.7-3.el7.remi.x86_64
php-fedora-autoloader-1.0.0-1.el7.noarch
php-symfony-finder-2.8.12-2.el7.noarch
php-symfony-css-selector-2.8.12-2.el7.noarch
php-symfony-debug-2.8.12-2.el7.noarch
php-symfony-var-dumper-2.8.12-2.el7.noarch
php-pdo-7.3.7-3.el7.remi.x86_64
php-xml-7.3.7-3.el7.remi.x86_64
php-symfony-common-2.8.12-2.el7.noarch
php-composer-ca-bundle-1.0.2-1.el7.noarch
php-symfony-polyfill-1.5.0-1.el7.noarch
php-mbstring-7.3.7-3.el7.remi.x86_64

Naemon -version
naemon-1.0.10-4.3.noarch

User is there
# ./bin/Console.php users
+-----------------------------------+--------------------------------------------------------------+
| Username                          | Password                                                     |
+-----------------------------------+--------------------------------------------------------------+
| admin                             | $2y$10$I4IUZDv1OJEeqXVM..ssMuQcbJqswPTabp5Cx2AUTGGe/hhfRn9hi |
| test                              | $2y$10$XTicdZlq5bi0qGzjl2TRA.je7FUT4CYfLuBhY5hI1.jlBGZqTi81e |
+-----------------------------------+--------------------------------------------------------------+

There is data in the mariadb database
MariaDB [statusengine]> select * from statusengine_hoststatus;
+-----------------+--------------------+--------------------------------------------+-------------+----------------------------------------------------------------+---------------+-----------------------+--------------------+------------+------------+------------------+-------------------+------------------------+-----------------+--------------+-------------------+-------------------+-----------------------+-------------------------------+----------------------+------------------------+-----------------------+-----------------------+------------------------+-------------+----------+----------------+--------------------------+--------------------------+------------------+-----------------------+----------------------+------------------+-----------+--------------+----------------+-----------------------+-----------------------------+----------------------+---------------+------------------+
| hostname        | status_update_time | output                                     | long_output | perfdata                                                       | current_state | current_check_attempt | max_check_attempts | last_check | next_check | is_passive_check | last_state_change | last_hard_state_change | last_hard_state | is_hardstate | last_notification | next_notification | notifications_enabled | problem_has_been_acknowledged | acknowledgement_type | passive_checks_enabled | active_checks_enabled | event_handler_enabled | flap_detection_enabled | is_flapping | latency  | execution_time | scheduled_downtime_depth | process_performance_data | obsess_over_host | normal_check_interval | retry_check_interval | check_timeperiod | node_name | last_time_up | last_time_down | last_time_unreachable | current_notification_number | percent_state_change | event_handler | check_command    |
+-----------------+--------------------+--------------------------------------------+-------------+----------------------------------------------------------------+---------------+-----------------------+--------------------+------------+------------+------------------+-------------------+------------------------+-----------------+--------------+-------------------+-------------------+-----------------------+-------------------------------+----------------------+------------------------+-----------------------+-----------------------+------------------------+-------------+----------+----------------+--------------------------+--------------------------+------------------+-----------------------+----------------------+------------------+-----------+--------------+----------------+-----------------------+-----------------------------+----------------------+---------------+------------------+
| XXX |         1564089797 | PING OK - Packet loss = 0%, RTA = 0.34 ms  | NULL        | rta=0.340000ms;3000.000000;5000.000000;0.000000 pl=0%;80;100;0 |             0 |                     1 |                  3 | 1564089793 | 1564090093 |                1 |        1564078697 |             1564078697 |               0 |            1 |                 0 |                 0 |                     1 |                             0 |                    0 |                      1 |                     1 |                     1 |                      1 |           0 |  0.25737 |        4.00456 |                        0 |                        1 |                1 |                     5 |                    1 | 24x7             | Crowbar   |   1564089797 |     1564078697 |                     0 |                           0 |                    0 | NULL          | check-host-alive |
| hplj2605dn      |         1564089908 | (Host check timed out after 30.00 seconds) | NULL        | NULL                                                           |             1 |                     1 |                 10 | 1564089878 | 1564090178 |                1 |        1562687695 |             1562687695 |               1 |            1 |        1563226300 |        1563228100 |                     1 |                             0 |                    0 |                      1 |                     1 |                     1 |                      1 |           0 | 0.257856 |        30.0014 |                        0 |                        1 |                1 |                     5 |                    1 | 24x7             | Crowbar   |   1562687180 |     1564089908 |                     0 |                          77 |                    0 | NULL          | check-host-alive |
| linksys-srw224p |         1564089952 | (Host check timed out after 30.00 seconds) | NULL        | NULL                                                           |             1 |                     1 |                 10 | 1564089922 | 1564090222 |                1 |        1562687683 |             1562687683 |               1 |            1 |        1563229672 |        1563231472 |                     1 |                             0 |                    0 |                      1 |                     1 |                     1 |                      1 |           0 | 0.264405 |        30.0014 |                        0 |                        1 |                1 |                     5 |                    1 | 24x7             | Crowbar   |   1562687173 |     1564089952 |                     0 |                         302 |                    0 | NULL          | check-host-alive |
| localhost       |         1564089922 | PING OK - Packet loss = 0%, RTA = 0.04 ms  | NULL        | rta=0.037000ms;3000.000000;5000.000000;0.000000 pl=0%;80;100;0 |             0 |                     1 |                 10 | 1564089918 | 1564090218 |                1 |        1564078822 |             1564078822 |               0 |            1 |                 0 |                 0 |                     1 |                             0 |                    0 |                      1 |                     1 |                     1 |                      1 |           0 | 0.258723 |        4.00374 |                        0 |                        1 |                1 |                     5 |                    1 | 24x7             | Crowbar   |   1564089922 |     1564078822 |                     0 |                           0 |                    0 | NULL          | check-host-alive |
| winserver       |         1564089926 | (Host check timed out after 30.00 seconds) | NULL        | NULL                                                           |             1 |                     1 |                 10 | 1564089896 | 1564090196 |                1 |        1562687624 |             1562687624 |               1 |            1 |        1563229482 |        1563231282 |                     1 |                             0 |                    0 |                      1 |                     1 |                     1 |                      1 |           0 | 0.263008 |        30.0016 |                        0 |                        1 |                1 |                     5 |                    1 | 24x7             | Crowbar   |   1562687104 |     1564089926 |                     0 |                         302 |                    0 | NULL          | check-host-alive |
+-----------------+--------------------+--------------------------------------------+-------------+----------------------------------------------------------------+---------------+-----------------------+--------------------+------------+------------+------------------+-------------------+------------------------+-----------------+--------------+-------------------+-------------------+-----------------------+-------------------------------+----------------------+------------------------+-----------------------+-----------------------+------------------------+-------------+----------+----------------+--------------------------+--------------------------+------------------+-----------------------+----------------------+------------------+-----------+--------------+----------------+-----------------------+-----------------------------+----------------------+---------------+------------------+
5 rows in set (0.00 sec)

MariaDB [statusengine]>
M

**Comment by nook24** [comment]
> Notice below error message in /var/log/httpd/
>AH00130: File does not exist: /usr/share/statusengine-ui/public/api/index.php/login

**Did you disabled SELinux?**
It caused issues on my CentOS testing system: https://github.com/statusengine/interface/issues/23#issuecomment-502144049

Maybe this helps you too: https://github.com/statusengine/interface/issues/23#issuecomment-502380455

**Comment by nishitjpatel** [comment]
Selinux is in permissive mode and I went through that post before opening the issue.  I am using redhat7.6,php73, and mariadb.  Anything else you can think of?    Thanks for your reply.

**Comment by nook24** [comment]
Did you checked that `mod_rewrite.so` got loaded properly?

This is the `/etc/httpd/conf.d/statusengine-ui.conf` for CentOS from the docs
```apache

<VirtualHost 192.168.56.101:80>
    ServerName statusengine.example.org

    RewriteEngine On
    RewriteCond %{HTTPS} off
    RewriteRule (.*) https://%{SERVER_NAME}/$1 [R,L]

    DocumentRoot "/usr/share/statusengine-ui/public/"

    RedirectMatch 404 /\.git

    ErrorLog "/var/log/apache2/statusengine-ui-error.log"
    CustomLog "/var/log/apache2/statusengine-ui-access.log" combined
</VirtualHost>

<VirtualHost 192.168.56.101:443>
    ServerName statusengine.example.org

    RedirectMatch 404 /\.git

    DocumentRoot "/usr/share/statusengine-ui/public/"

    SSLEngine On
    SSLCertificateFile    /etc/ssl/certs/ssl-cert-snakeoil.pem
    SSLCertificateKeyFile /etc/ssl/private/ssl-cert-snakeoil.key

    ErrorLog "/var/log/apache2/statusengine-ui-error.log"
    CustomLog "/var/log/apache2/statusengine-ui-access.log" combined
</VirtualHost>
```

And how to load mod_rewirte and enable the configuration...
```
rm /etc/httpd/conf.d/welcome.conf

mkdir /var/log/apache2

echo "LoadModule rewrite_module modules/mod_rewrite.so" >> /etc/httpd/conf.modules.d/00-base.conf

echo "<Directory /usr/share/statusengine-ui/public>" >> /etc/httpd/conf.d/statusengine-ui.conf
echo "  Require all granted" >> /etc/httpd/conf.d/statusengine-ui.conf
echo "</Directory>" >> /etc/httpd/conf.d/statusengine-ui.conf

systemctl restart httpd
```

Hope this helps

**Comment by nishitjpatel** [comment]
that's all set in place and i have restarted httpd several times 

<VirtualHost XXX:80>
    ServerName XXX

    RewriteEngine On
    RewriteCond %{HTTPS} off
    RewriteRule (.*) https://%{SERVER_NAME}/$1 [R,L]

    DocumentRoot "/usr/share/statusengine-ui/public/"

    RedirectMatch 404 /\.git

    ErrorLog "/var/log/httpd/statusengine-ui-error.log"
    CustomLog "/var/log/httpd/statusengine-ui-access.log" combined
</VirtualHost>

<VirtualHost XXX:443>
    ServerName XXX

    RedirectMatch 404 /\.git

    DocumentRoot "/usr/share/statusengine-ui/public/"

    SSLEngine On
    SSLCertificateFile    /etc/ssl/certs/http-statusengine.crt
    SSLCertificateKeyFile /etc/ssl/certs/http-statusengine.key


    ErrorLog "/var/log/httpd/statusengine-ui-error.log"
    CustomLog "/var/log/httpd/statusengine-ui-access.log" combined
</VirtualHost>
<Directory /usr/share/statusengine-ui/public>
  AllowOverride All
  Require all granted
</Directory>

# ls -l /etc/httpd/conf.d/welcome.conf*
-rw-r--r--. 1 root root 516 Mar 15 06:15 /etc/httpd/conf.d/welcome.conf~

# ls -ld /var/log/httpd
drwx------. 2 root root 4096 Jul 26 03:08 /var/log/httpd

# grep rewrite /etc/httpd/conf.modules.d/00-base.conf
LoadModule rewrite_module modules/mod_rewrite.so

**Comment by nishitjpatel** [comment]
sorry should have config in the code insert
`
# cat /etc/httpd/conf.d/statusengine-ui.conf
<VirtualHost XXX:80>
    ServerName XXX

    RewriteEngine On
    RewriteCond %{HTTPS} off
    RewriteRule (.*) https://%{SERVER_NAME}/$1 [R,L]

    DocumentRoot "/usr/share/statusengine-ui/public/"

    RedirectMatch 404 /\.git

    ErrorLog "/var/log/httpd/statusengine-ui-error.log"
    CustomLog "/var/log/httpd/statusengine-ui-access.log" combined
</VirtualHost>

<VirtualHost XXX:443>
    ServerName XXX

    RedirectMatch 404 /\.git

    DocumentRoot "/usr/share/statusengine-ui/public/"

    SSLEngine On
    SSLCertificateFile    /etc/ssl/certs/http-statusengine.crt
    SSLCertificateKeyFile /etc/ssl/certs/http-statusengine.key


    ErrorLog "/var/log/httpd/statusengine-ui-error.log"
    CustomLog "/var/log/httpd/statusengine-ui-access.log" combined
</VirtualHost>
<Directory /usr/share/statusengine-ui/public>
  AllowOverride All
  Require all granted
</Directory>

`

**Comment by nishitjpatel** [comment]
I am baffled but it works today and i have made zero change.    I logged into check and my httpd daemon was dead.. i restarted it and it all now works just fine.    I am 100% sure i restarted httpd 10-15 times yesterday experimenting with config.yml for statusUI.     

We can close this issue.    Sorry and thank you for your time

**Comment by nook24** [comment]
🤷‍♂️
I‘m happy that it‘s working. You are welcome
